File: jandy-python/jandy/ThreadObject.py
from jandy.Serializer import Serializer


class ThreadObject(Serializer):
    def __init__(self):
        self.threadId = ""
        self.threadName = ""
        self.rootId = ""

    def getThreadName(self):
        return self.threadName

    def setThreadName(self, threadName):
        self.threadName = threadName
        return self

    def getThreadId(self):
        return self.threadId

    def setThreadId(self, threadId):
        self.threadId = threadId
        return self

    def getRootId(self):
        return self.rootId

    def setRootId(self, rootId):
        self.rootId = rootId
        return self

    def toDict(self):
        return {
            'threadId': self.threadId,
            'threadName': self.threadName,
            'rootId': self.rootId
        }

File: jandy-java-profiler/src/main/java/io/jandy/java/data/ThreadObject.java
package io.jandy.java.data;

/**
 * @author JCooky
 * @since 2016-06-30
 */
public class ThreadObject {
  private long threadId;
  private String threadName;
  private String rootId;

  public String getThreadName() {
    return threadName;
  }

  public ThreadObject setThreadName(String threadName) {
    this.threadName = threadName;
    return this;
  }

  public long getThreadId() {
    return threadId;
  }

  public ThreadObject setThreadId(long threadId) {
    this.threadId = threadId;
    return this;
  }

  public String getRootId() {
    return rootId;
  }

  public ThreadObject setRootId(String rootId) {
    this.rootId = rootId;
    return this;
  }

}
